codec-http/src/main/java/io/netty/handler/codec/http/HttpServerCodec.java

```diff
@@ -111,9 +111,25 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> ou
 
     private final class HttpServerResponseEncoder extends HttpResponseEncoder {
 
+        private HttpMethod method;
+
+        @Override
+        protected void filterHeaders(HttpResponse msg, boolean isAlwaysEmpty) {
+            if (!isAlwaysEmpty) {
+                if (method == HttpMethod.CONNECT && msg.status().codeClass() == HttpStatusClass.SUCCESS) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)